src/main/java/gregtech/integration/jei/utils/JEIResourceDepositCategoryUtils.java

```diff
@@ -0,0 +1,157 @@
+package gregtech.integration.jei.utils;
+
+import gregtech.api.util.GTLog;
+import gregtech.api.worldgen.config.OreDepositDefinition;
+import it.unimi.dsi.fastutil.ints.IntArrayList;
+import it.unimi.dsi.fastutil.ints.IntList;
+import it.unimi.dsi.fastutil.ints.IntSortedSet;
+import it.unimi.dsi.fastutil.objects.Object2IntMap;
+import it.unimi.dsi.fastutil.objects.Object2IntOpenHashMap;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.world.DimensionType;
+import net.minecraft.world.WorldProvider;
+import net.minecraft.world.biome.Biome;
+import net.minecraftforge.common.DimensionManager;
+import net.minecraftforge.fml.common.Loader;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import static gregtech.api.GTValues.MODID_AR;
+
+/**
+ * Common util methods shared between {@link gregtech.integration.jei.basic.GTOreCategory}
+ * and {@link gregtech.integration.jei.basic.GTFluidVeinCategory}
+ */
+public class JEIResourceDepositCategoryUtils {
+
+    /**
+     * Creates a list of biomes whose weight, as described by the passed Function, differs from the passed original weight
+     * For use in the JEI Ore Spawn Page and JEI Fluid Spawn Page
+     *
+     * @param biomeFunction  A Function describing the modified weights of biomes
+     * @param originalWeight The original weight of the biome
+     * @return A List containing all the modified biomes and their value they are modified by
+     */
+    public static List<String> createSpawnPageBiomeTooltip(Function<Biome, Integer> biomeFunction, int originalWeight) {
```

Comment:
Use `Object2IntFunction` instead?

Reply:
The `Function<Biome, Integer>` originates from part of the ore/fluid vein API.